stock_secondary_unit/models/stock_move.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_round
+
+
+class StockMove(models.Model):
+    _inherit = ["stock.move", "product.secondary.unit.mixin"]
+    _name = "stock.move"
+    _secondary_unit_fields = {
+        "qty_field": "product_uom_qty",
+        "uom_field": "product_uom",
+    }
+
+    product_uom_qty = fields.Float(
+        store=True,
+        readonly=False,
+        compute="_compute_product_uom_qty",
+        copy=True,
+        precompute=True,
+    )
+
+    @api.depends("secondary_uom_qty", "secondary_uom_id")
+    def _compute_product_uom_qty(self):
+        self._compute_helper_target_field_qty()
+
+    @api.onchange("product_uom")
+    def onchange_product_uom_for_secondary(self):
+        self._onchange_helper_product_uom_for_secondary()
+
+    @api.model
+    def _prepare_merge_moves_distinct_fields(self):
+        """Don't merge moves with distinct secondary units"""
+        distinct_fields = super()._prepare_merge_moves_distinct_fields()
+        distinct_fields += ["secondary_uom_id"]
+        return distinct_fields
+
+
```

Comment:
This ensures calculation of secondary qty. See comments in main thread.

```suggestion
    @api.model_create_multi
    def create(self, vals_list):
        for vals in vals_list:
            product = self.env["product.product"].browse(vals.get("product_id", False))
            if product:
                vals.update(
                    {
                        "secondary_uom_id": product.stock_secondary_uom_id.id
                        or product.product_tmpl_id.stock_secondary_uom_id.id,
                    }
                )
        return super().create(vals_list)
```

Reply:
Hello, I'm in a similar situation to yours. After reviewing [OCA/product-attribute#377](https://github.com/OCA/product-attribute/pull/377), I gained some insights, but I still suspect that we might both have incorrect expectations. For example, your suggestion doesn't account for the fact that secondary unit fields can be optional.

I believe the issue lies here:  
- [`product_secondary_unit_mixin.py#L38`](https://github.com/OCA/product-attribute/blob/439e1559e873d0445c323e5a31f0360fdaa88e46/product_secondary_unit/models/product_secondary_unit_mixin.py#L38)  
- [`product_template.py#L18-L23`](https://github.com/OCA/product-attribute/blob/439e1559e873d0445c323e5a31f0360fdaa88e46/product_secondary_unit/models/product_template.py#L18-L23)  

The use of `@api.model` ensures that `self` will always be the model, not a record. Consequently, `secondary_oum_ids` will always evaluate to `None`, which might be causing the problem.

The fact that everyone here seems to ignore your concerns makes me think we might be misunderstanding the addon's purpose or intended usage.

ping: @sergio-teruel @lk-eska @rousseldenis @jbaudoux :face_with_head_bandage:

Reply:
Take a look at this: [Commit 35374e39586c](https://github.com/OCA/stock-logistics-warehouse/commit/35374e39586c53b8fa6933f393eae7e0e2d06e54). There's no doubt the behavior is intentional. Why? 🤔 ... That was never explained.